src/test/java/com/bakdata/kafka/util/SchemaTopicClientTest.java

```diff
@@ -47,64 +48,129 @@
                 .as("Topic is created")
                 .isTrue();
 
         final SendValuesTransactional<TestRecord> sendRequest = SendValuesTransactional
                 .inTransaction(TOPIC, List.of(TestRecord.newBuilder().setContent("foo").build()))
                 .with(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, SpecificAvroSerializer.class)
-                .with(AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, this.schemaRegistryMockExtension.getUrl())
+                .with(AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG,
+                        this.schemaRegistryMockExtension.getUrl())
                 .build();
         this.kafkaCluster.send(sendRequest);
 
         final SchemaRegistryClient client = this.schemaRegistryMockExtension.getSchemaRegistryClient();
-        assertThat(client.getAllSubjects())
+        this.softly.assertThat(client.getAllSubjects())
                 .contains(TOPIC + "-value");
 
-        try (final SchemaTopicClient schemaTopicClient = this.createSchemaTopicClient()) {
+        try (final SchemaTopicClient schemaTopicClient = this.createClientWithSchemaRegistry()) {
             schemaTopicClient.deleteTopicAndResetSchemaRegistry(TOPIC);
         }
 
         delay(TIMEOUT_SECONDS, TimeUnit.SECONDS);
-        assertThat(this.kafkaCluster.exists(TOPIC))
-                .as("Topic is deleted")
+
+        this.softly.assertThat(client.getAllSubjects())
+                .doesNotContain(TOPIC + "-value");
+        this.softly.assertThat(this.kafkaCluster.exists(TOPIC))
                 .isFalse();
-        assertThat(client.getAllSubjects())
+    }
+
+    @Test
+    void shouldDeleteTopic() throws InterruptedException, IOException, RestClientException {
+        this.kafkaCluster.createTopic(TopicConfig.withName(TOPIC).useDefaults());
+        this.softly.assertThat(this.kafkaCluster.exists(TOPIC))
+                .as("Topic is created")
+                .isTrue();
+
+        final SendValuesTransactional<TestRecord> sendRequest = SendValuesTransactional
+                .inTransaction(TOPIC, List.of(TestRecord.newBuilder().setContent("foo").build()))
+                .with(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, SpecificAvroSerializer.class)
+                .with(AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG,
+                        this.schemaRegistryMockExtension.getUrl())
+                .build();
+        this.kafkaCluster.send(sendRequest);
+
+        final SchemaRegistryClient client = this.schemaRegistryMockExtension.getSchemaRegistryClient();
+        this.softly.assertThat(client.getAllSubjects())
+                .contains(TOPIC + "-value");
+
+        try (final SchemaTopicClient schemaTopicClient = this.createClientWithSchemaRegistry()) {
+            schemaTopicClient.resetSchemaRegistry(TOPIC);
+        }
+
+        delay(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        this.softly.assertThat(client.getAllSubjects())
```

Comment:
Isn't that the same test as above?

Reply:
One is testing `deleteTopicAndResetSchemaRegistry` method and the other one this `resetSchemaRegistry`. I improved the naming of the methods and the test assertion